### PR TITLE
Remove duplicated map serialization code

### PR DESCRIPTION
**Pull request by JackPGreen**
`SerializationUtil` contains a utility method to deser a map with `Object` keys, by deser it's size and then entry contents.

An identical scenario occurs for maps with `String` keys, but this implementation is duplicated across the system - I've changed this so it's centralised also.

EE PR: https://github.com/hazelcast/hazelcast-enterprise/pull/6658

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/internal/crdt/AbstractCRDTReplicationOperation.java [hunk 1]
```diff
@@ -78,21 +76,11 @@ public int getFactoryId() {
...
     @Override
     protected void readInternal(ObjectDataInput in) throws IOException {
-        final int mapSize = in.readInt();
-        replicationData = createHashMap(mapSize);
-        for (int i = 0; i < mapSize; i++) {
-            final String name = in.readString();
-            final T crdt = in.readObject();
-            replicationData.put(name, crdt);
-        }
+        replicationData = SerializationUtil.readMap(in);
```
Comment (vbekiaris) [inline hazelcast/src/main/java/com/hazelcast/internal/crdt/AbstractCRDTReplicationOperation.java:84]:
```suggestion
        replicationData = SerializationUtil.readMapStringKey(in);
```

File: hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ConfigCheck.java [hunk 2]
```diff
@@ -196,13 +187,8 @@ public void readData(ObjectDataInput in) throws IOException {
...
         }
-        int propSize = in.readInt();
-        properties = createHashMap(propSize);
-        for (int k = 0; k < propSize; k++) {
-            String key = in.readString();
-            String value = in.readString();
-            properties.put(key, value);
-        }
+
+        properties = SerializationUtil.readMap(in);
```
Comment (vbekiaris) [inline hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ConfigCheck.java:191]: this change should be reverted - `readObject` (used in `readMap`) does not read the same serialized form as the one produced by `writeString` on the `writeData` side

Comment (JackPGreen) [reply hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ConfigCheck.java]: a5710bbe37ed471893b68af41ed0e36447bcdfe5

File: hazelcast/src/main/java/com/hazelcast/internal/crdt/AbstractCRDTReplicationOperation.java (hunk 1 above)
Comment (vbekiaris) [reply hazelcast/src/main/java/com/hazelcast/internal/crdt/AbstractCRDTReplicationOperation.java:84]: looks like this change was reverted?

**Review by vbekiaris — APPROVED** [review]

File: hazelcast/src/main/java/com/hazelcast/internal/crdt/AbstractCRDTReplicationOperation.java (hunk 1 above)
Comment (JackPGreen) [reply hazelcast/src/main/java/com/hazelcast/internal/crdt/AbstractCRDTReplicationOperation.java]: > looks like this change was reverted?

It's definitely there, GitHub just isn't showing it in the PR diff for some reason?

https://github.com/JackPGreen/hazelcast/blob/remove-replicate-map-serialization-code/hazelcast/src/main/java/com/hazelcast/internal/crdt/AbstractCRDTReplicationOperation.java#L79

**Review by JamesHazelcast — APPROVED** [review]

**Pull request merged**